src/main/java/frc/team166/robot/subsystems/Manipulator.java

```diff
@@ -0,0 +1,246 @@
+		addChild(rollers);
+		addChild(irSensor);
+		addChild(leftIntakeEncoder);
+		addChild(rightIntakeEncoder);
+
+		leftIntakeEncoder.setReverseDirection(true);
+		leftRoller.setInverted(true);
+
+		leftIntakeEncoder.setDistancePerPulse(DIST_PER_PULSE_INTAKE);
+		rightIntakeEncoder.setDistancePerPulse(DIST_PER_PULSE_INTAKE);
+	}
+
+	/**
+	 * Resets encoders
+	 * <p>
+	 * Calls the reset function on the left and right intake encoders
+	 */
+	public void resetEncoders() {
+		leftIntakeEncoder.reset();
+		rightIntakeEncoder.reset();
+	}
+
+	/**
+	 * Gets IR voltage
+	 * <p>
+	 * Returns the voltage value output from the IR sensor
+	 * 
+	 * @return The voltage from the IR sensor
+	 */
+	public double getIRDistance() {
+		return irSensor.getVoltage(); //Manipulate this data pending experimentation
+	}
+
+	public void openManipulator() {
+		manipulatorSolenoid.set(Value.kForward);
+		;
+	}
+
+	public void closeManipulator() {
+		manipulatorSolenoid.set(Value.kReverse);
+	}
+
+	public double getAvgEncoderRate() { //ft/s
+		double leftRate = leftIntakeEncoder.getRate();
+		double rightRate = rightIntakeEncoder.getRate();
+		return (leftRate + rightRate) / 2.0;
+	}
+
+	/**
+	 * Gets average encoder distance
+	 * <p>
+	 * Returns the average distance of the left and right encoders
+	 * 
+	 * @return The average distance of the two encoders
+	 */
+	public double getAvgEncoderDistance() { //ft
+		double leftDist = leftIntakeEncoder.getDistance();
+		double rightDist = rightIntakeEncoder.getDistance();
+		return (leftDist + rightDist) / 2.0;
+	}
+
+	/**
+	 * Sets motors to intake mode
+	 * <p>
+	 * Sets the motors to 4/5 power inward to take in a cube on the field
+	 */
+	public void setMotorsToIntake() {
+		rollers.set(0.8); //change once you find optimal motor speed
+	}
+
+	/**
+	 * Sets motors to discharge mode
+	 * <p>
+	 * Sets the motors to 4/5 power outward to eject a stored cube
+	 */
+	public void setMotorsToDischarge() {
+		rollers.set(-0.8); //change once you find optimal motor speed
+	}
+
+	//COMMANDS
+	public void initDefaultCommand() {
+		setDefaultCommand(new SubsystemCommand(this) {
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+		});
+	}
+
+	public Command DropCube() {
+		return new SubsystemCommand(this) {
+
+			@Override
+			protected void initialize() {
+				openManipulator();
+			}
+
+			@Override
+			protected void execute() {
+
+			}
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+			@Override
+			protected void end() {
+
+			}
+		};
+	}
+
+	public Command CloseManipulator() {
+		return new SubsystemCommand(this) {
+
+			@Override
+			protected void initialize() {
+				closeManipulator();
+			}
+
+			@Override
+			protected void execute() {
+
+			}
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+			@Override
+			protected void end() {
+
+			}
+		};
+	}
+
+	public Command CubePickup() {
+		return new SubsystemCommand(this) {
```

Comment:
Use `new SubsystemCommand("Pick Up Cube", this)` instead of just `this`, to give it a name that shows up nicely on smart dashboard